### PR TITLE
Update attrs to 21.2.0

### PR DESCRIPTION
**Pull request by davfsa**
- Use next generation API
 - Cleanup redundant arguments

### Checklist

- [x] I have run `nox` and all the pipelines have passed.
- [x] I have made unittests according to the code I have added/modified/deleted.

### CODE REVIEW

**Comment by FasterSpeeding** [comment]
What's the point of most of these changes, what does this bring or avoid?

**Comment by davfsa** [comment]
> What's the point of most of these changes, what does this bring or avoid?

Using the next generation API is preferred, as well as being nicer to work with and read.  

References:
[1] https://www.attrs.org/en/stable/api.html (note box)
[2] https://www.attrs.org/en/stable/api.html#next-gen

File: hikari/api/entity_factory.py [hunk 1]
```diff
@@ -54,21 +54,21 @@
...
-    channels: typing.Optional[typing.Mapping[snowflakes.Snowflake, channel_models.GuildChannel]] = attr.ib()
+    channels: typing.Optional[typing.Mapping[snowflakes.Snowflake, channel_models.GuildChannel]]
     """Mapping of channel IDs to the channels that belong to the guild.
 
     Will be `builtins.None` when returned by guild update gateway events rather
     than create.
     """
 
-    members: typing.Optional[typing.Mapping[snowflakes.Snowflake, guild_models.Member]] = attr.ib()
+    members: typing.Optional[typing.Mapping[snowflakes.Snowflake, guild_models.Member]]
```
Comment (FasterSpeeding) [inline hikari/api/entity_factory.py:71]: Seems a bit inconsistent to reintroduce this behaviour where we arbitrarily do or don't use auto_attribs in random places

Comment (FasterSpeeding) [reply hikari/api/entity_factory.py:71]: LGTM otherwise

**Review by FasterSpeeding — APPROVED** [review]

**Pull request merged**